src/base32/alphabet.rs

```diff
@@ -215,20 +214,32 @@ impl Alphabet32Padded {
     /// Pass decoder array to [`decode`](super::decode()), and remove padding as needed
     #[inline]
     pub fn decode(&self, a: &[u8]) -> Result<Vec<u8>, DecodeError> {
-        decode(self.dec, rem_pad(a, self.pad))
+        if a.len() % WIDTH_ENC != 0 {
```

Comment:
New check: make sure this is a valid padded length (still does not verify that it is actually a padded string! -- that will fail for invalid char, or actually parse if it is valid)